public/scripts/main.js

```diff
@@ -56,16 +56,60 @@ function replaceWithInput(type, index) {
   }
 }
 
+function validateFields(bookNumber){
+  let exitCode = false
+  for(let field of fields){
+    const domField = document.querySelector('#'+field+'Input'+bookNumber)
+    let colorThisDom = false
+
+    if(domField){
+      if(domField.value){
+        // Note to PURR reviewer:
+        // This conditional cannot be rewritten as domField.value.length <= 0
+        // because currently, this conditional reverses any NaN possibilities.
+
+        // That is: !(NaN > 0) yields true
+        // whereas (NaN <= 0) yields false
+        // and we need non number lengths to be perfectly okay and valid.
+        const value = !(domField.value.length > 0)
```

Comment:
The constants on lines 74 and 78 aren't necessary since they're just being used to assign the boolean to exitCode.

if(!exitCode) exitCode = !(domField.value.length > 0)